### ISSUE TITLE
Create routing so each page has a separate url

### ISSUE THREAD

**Issue by ayya-vimala** [ayya-vimala/pali-networks#28]
A url per page and text would be helpful.
F.i. `buddhist-db.de/pali/mn2` or `buddhist-db.de/pali?text=mn2`
This would help with linking up elements and would make it easier for users to revisit certain pages.

This would require setting up routing as opposed to lazy loading.
This is a fairly big thing to do that impacts the whole site. The drawback would be that the menu is loaded again each time a page is loaded which takes a long time.

Possible solutions: 
* enter the menus into Redux store
* use a backend system like MongoDB to serve menus

Might need the help of a professional developer for this.

**Comment by ayya-vimala** [comment]
@sebastian-nehrdich 
I've created a branch called 'router-setup'. The CSS for the links is not that great but otherwise it sort of works. In principle I would also like it to work with a route like /tibetan/T06TD4021E but did not get that far yet.
However, this looses the ability to keep pages of another language you might also be looking at (as with the iron-pages). So question is: do we want that or do we want to keep the iron-pages and forget about routing?
The one thing is of course that people can also open mulitple browser tabs or windows to see both chinese and tibetan at the same time so in that way routing might be better.

**Comment by sebastian-nehrdich** [comment]
My intuition would be to go for routing rather than for iron pages. That
will also make it possible for people the share links to files directly,
which in study group/classroom situations is always quite important I
think.
Even though I think the iron pages thing is quite cool, people will
intuitively go for different tabs rather than using different languages at
the same time. Just thinking because most people will focus on one text at
one time probably, or might look at another text in the same language than
in a different language. That's just my quick thought on this!

**Comment by ayya-vimala** [comment]
I tried two different routers that both should work but somehow they don't work very well on the local development. I reverted back to Vaadin as this is the easiest one. 
Problem is that if you have a direct link (in the current load of the router-branch the Tibetan tab opens a file straight away as an example), it works just fine. But if you type the same in the browser window, it does not work and fails with an error:

> GET http://localhost:8080/tibetan/main.77f9a805.js net::ERR_ABORTED 404 (Not Found)

It does this on both routers I tried so the problem must be elsewhere.

As I cannot try it out on the main server, I'm a bit of a loss what to do now.
I tried it once on the main server but then it wanted to go back to the root and not the `/vimala` so that did not work but I could not try much because the server was needed by Sebastian.

It might be something very simple.
